--- xdg-utils-1.1.1/scripts/xdg-screensaver.in	2015-10-05 11:59:19.000000000 -0700
+++ fixed/scripts/xdg-screensaver.in	2015-12-22 19:00:51.255340864 -0800
@@ -372,16 +372,29 @@
     esac
 }
 
+xset_screensaver_timeout()
+{
+  xset q | sed '/^Screen Saver:/,/^[^ ]/ { s/.*timeout: *\([0-9]*\).*/\1/; t }; d'
+}
+
 screensaver_xserver()
 {
     case "$1" in
         suspend)
-        xset s off > /dev/null
+        timeout=`xset_screensaver_timeout`
+        if [ "$timeout" -gt 0 ]; then
+            echo "$timeout" > "$screensaver_file.xset"
+            xset s off > /dev/null
+        fi
         result=$?
         ;;
 
         resume)
-        xset s default > /dev/null
+        if [ -f "$screensaver_file.xset" ] ; then
+            value=`cat "$screensaver_file.xset"`
+            xset s $value > /dev/null
+            rm -f "$screensaver_file.xset"
+        fi
         result=$?
         ;;
 
@@ -396,7 +409,7 @@
         ;;
 
         status)
-        timeout=`xset q | sed '/^Screen Saver:/,/^[^ ]/ { s/.*timeout: *\([0-9]*\).*/\1/; t }; d'`
+        timeout=`xset_screensaver_timeout`
         result=$?
         if [ "$timeout" -gt 0 ]; then
             echo "enabled"
